control-plane/subcommand/common/common.go

```diff
@@ -86,60 +85,72 @@ func ValidateUnprivilegedPort(flagName, flagValue string) error {
 // The logic of this is taken from the `consul login` command.
-func ConsulLogin(client *api.Client, cfg *api.Config, authMethodName, datacenter, namespace, bearerTokenFile, serviceAccountName, tokenSinkFile string, meta map[string]string, log hclog.Logger) (string, error) {
+func ConsulLogin(client *api.Client, params LoginParams, log hclog.Logger) (string, error) {
 	// Read the bearerTokenFile.
-	data, err := ioutil.ReadFile(bearerTokenFile)
+	data, err := ioutil.ReadFile(params.BearerTokenFile)
 	if err != nil {
-		return "", fmt.Errorf("unable to read bearerTokenFile: %v, err: %v", bearerTokenFile, err)
+		return "", fmt.Errorf("unable to read bearer token file: %v, err: %v", params.BearerTokenFile, err)
 	}
 	bearerToken := strings.TrimSpace(string(data))
 	if bearerToken == "" {
-		return "", fmt.Errorf("no bearer token found in %s", bearerTokenFile)
+		return "", fmt.Errorf("no bearer token found in %q", params.BearerTokenFile)
+	}
+
+	if params.numRetries == 0 {
+		params.numRetries = numLoginRetries
 	}
+	var token *api.ACLToken
 	err = backoff.Retry(func() error {
 		// Do the login.
 		req := &api.ACLLoginParams{
-			AuthMethod:  authMethodName,
+			AuthMethod:  params.AuthMethod,
 			BearerToken: bearerToken,
-			Meta:        meta,
+			Meta:        params.Meta,
 		}
 		// The datacenter flag will either have the value of the primary datacenter or "". In case of the latter,
 		// the token will be created in the datacenter of the installation. In case a global token is required,
 		// the token will be created in the primary datacenter.
-		tok, _, err := client.ACL().Login(req, &api.WriteOptions{Namespace: namespace, Datacenter: datacenter})
+		token, _, err = client.ACL().Login(req, &api.WriteOptions{Namespace: params.Namespace, Datacenter: params.Datacenter})
 		if err != nil {
 			log.Error("unable to login", "error", err)
 			return fmt.Errorf("error logging in: %s", err)
 		}
-		if tokenSinkFile != "" {
+		if params.TokenSinkFile != "" {
 			// Write out the resultant token file.
 			// Must be 0644 because this is written by the consul-k8s user but needs
 			// to be readable by the consul user
-			if err := WriteFileWithPerms(tokenSinkFile, tok.SecretID, 0644); err != nil {
+			if err = WriteFileWithPerms(params.TokenSinkFile, token.SecretID, 0644); err != nil {
 				return fmt.Errorf("error writing token to file sink: %v", err)
 			}
 		}
 		return err
-	}, backoff.WithMaxRetries(backoff.NewConstantBackOff(1*time.Second), numLoginRetries))
+	}, backoff.WithMaxRetries(backoff.NewConstantBackOff(1*time.Second), params.numRetries))
 	if err != nil {
-		if serviceAccountName == "default" {
-			log.Warn("The service account name for this Pod is \"default\"." +
-				" In default installations this is not a supported service account name." +
-				" The service account name must match the name of the Kubernetes Service" +
-				" or the consul.hashicorp.com/connect-service annotation.")
-		}
```

Comment:
I've moved this back into `connect-init` as I think it's only relevant there, and so it doesn't make sense to put it in `common`.